server/src/main/java/org/elasticsearch/common/bytes/BytesReference.java

```diff
@@ -74,6 +74,29 @@ static ByteBuffer[] toByteBuffers(BytesReference reference) {
         }
     }
 
+    /**
+     * Allocates new buffer and copy bytes from given BytesReference.
+     *
+     * @deprecated copying bytes is a right place for performance regression and unnecessary allocations.
+     * This method exists to serve very few places that struggle to handle reference counted buffers.
+     */
+    @Deprecated
```

Comment:
nit:

```suggestion
    @Deprecated(forRemoval = true)
```

And I agree with Lorenzo, maybe we should add it to `build-tools-internal/src/main/resources/forbidden/es-server-signatures.txt` so that you have to `@SuppressForbidden` in order to use it at all.

Reply:
Added to es-server-signatures.txt